plugins/modules/vmware_guest_disk.py

```diff
@@ -539,7 +588,7 @@ def get_sharing(self, disk, disk_type, disk_index):
 
         """
         sharing = disk.get('sharing')
-        if sharing and disk_type != 'eagerzeroedthick':
+        if sharing and disk_type != 'eagerzeroedthick' and disk_type != 'rdm':
```

Comment:
This can also be added in the doc of parameter 'sharing'.

Reply:
I have added note in the sharing parameter that rdm is not supported.